pkg/driver/driver_azure.go

```diff
@@ -150,13 +151,20 @@ func (d *AzureDriver) getVMParameters(vmName string, networkInterfaceReferenceID
 					},
 				},
 			},
-			AvailabilitySet: &compute.SubResource{
-				ID: &d.AzureMachineClass.Spec.Properties.AvailabilitySet.ID,
-			},
 		},
 		Tags: tagList,
 	}
 
+	if d.AzureMachineClass.Spec.Properties.Zone != nil {
+		VMParameters.Zones = &[]string{strconv.Itoa(*d.AzureMachineClass.Spec.Properties.Zone)}
+	}
+
+	if d.AzureMachineClass.Spec.Properties.AvailabilitySet != nil {
```

Comment:
Hi @dkistner , 

One small change. Can we move this to `else if`. As with validation, this shouldn't be a problem. However, if anyone is just importing this package it could be a problem.

Reply:
yep makes sense done!